packages/template-retail-react-app/app/hooks/use-active-data.js

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+/*global dw*/
+import {ACTIVE_DATA_ENABLE} from '@salesforce/retail-react-app/app/constants'
+
+const useActiveData = () => {
+    return {
+        async sendViewProduct(category, product, type) {
+            if (!ACTIVE_DATA_ENABLE) {
+                return
+            }
+            try {
+                if (dw?.ac) {
```

Comment:
```suggestion
            if (typeof dw === 'undefined') {
                console.warn('DW analytics not loaded')
                return
            }
            try {
                if (dw.ac) {
```
`dw?.ac` will still throw an error if the global variable has not been set.

Reply:
@bfeister mentioned that existing check should cover checking for dw as well